storage/client/README.md

```diff
@@ -0,0 +1,358 @@
+
+```Powershell
+az group create \
+    --name storage-resource-group \
+    --location westus
+```
+
+### Authenticate the client
+
+In order to interact with the Storage service (Blob, Queue, Message, MessageId, File) you'll need to create an instance of the Service Client class. 
+To make this possible you'll need the Account SAS (shared access signature) string of Storage account. Learn more at [SAS Token][sas_token]
+
+#### Get Credentials
+
+- SAS Token
+ 
+a. Use the [Azure CLI][azure_cli] snippet below to get the SAS token from the Storage account.
+
+```Powershell
+az storage queue generate-sas
+    --name {queue name}
+    --expiry {date/time to expire SAS token}
+    --permission {permission to grant}
+    --connection-string {connection string of the storage account}
+```
+
+```Powershell
+#usage example
+az storage queue generate-sas
+    --name javasdksas
+    --expiry 2019-06-05
+    --permission rpau
+    --connection-string DefaultEndpointsProtocol=https;AccountName=azstoragesdkaccount;AccountKey=PhOFTkuqB09cmsy0aMgd+cJfp75xU3dgjRMKrWxHHx4pFvuFIMOSK/fhzGci6zxFGk4CkmgoXFO3s91S9m+ICw==;EndpointSuffix=core.windows.net
+```
+b. Alternatively, get the Account SAS Token from the Azure Portal.
+```
+Go to your storage account -> Shared access signature -> Click on Generate SAS and connection string (after setup)
+```
+
+- Shared Key Credential
+a. Use account name and account key. Account name is your storage account name.
+```
+// Here is where we get the key
+Go to your storage account -> Access keys -> Key 1/ Key 2 -> Key
+```
+b. Use the connection string
+```
+// Here is where we get the key
+Go to your storage account -> Access Keys -> Keys 1/ Key 2 -> Connection string
+```
+## Key concepts
+### URL format
+Queues are addressable using the following URL format:
+https://<storage account>.queue.core.windows.net/<queue>
+The following URL addresses a queue in the diagram:
+https://myaccount.queue.core.windows.net/images-to-download
+
+#### Resource URI Syntax
+For the storage account, the base URI for queue operations includes the name of the account only:
+```$xslt
+https://myaccount.queue.core.windows.net
+```
+For a queue, the base URI includes the name of the account and the name of the queue:
+```$xslt
+https://myaccount.queue.core.windows.net/myqueue
+```
+
+### Queue Names
+Every queue within an account must have a unique name. The queue name must be a valid DNS name, and cannot be changed once created. Queue names must confirm to the following rules:
+1. A queue name must start with a letter or number, and can only contain letters, numbers, and the dash (-) character.
+1. The first and last letters in the queue name must be alphanumeric. The dash (-) character cannot be the first or last character. Consecutive dash characters are not permitted in the queue name.
+1. All letters in a queue name must be lowercase.
+1. A queue name must be from 3 through 63 characters long.
+
+### Metadata Names
```

Comment:
I'm not sure if this section needs to be included, this is more advanced features of queue service that don't need to be called out for SDK usage.

Reply:
Will consult with Ted, and update whether we should remove here.

Reply:
Ted also suggested we remove the details from high level readme and add comments in Sample code.